homeassistant/components/group/__init__.py

```diff
@@ -636,17 +658,19 @@ def extra_state_attributes(self):
         return data
 
     @property
-    def assumed_state(self):
+    def assumed_state(self) -> bool:
         """Test if any member has an assumed state."""
         return self._assumed_state
 
-    def update_tracked_entity_ids(self, entity_ids):
+    def update_tracked_entity_ids(self, entity_ids: Iterable[str] | None) -> None:
         """Update the member entity IDs."""
         asyncio.run_coroutine_threadsafe(
             self.async_update_tracked_entity_ids(entity_ids), self.hass.loop
         ).result()
 
-    async def async_update_tracked_entity_ids(self, entity_ids):
+    async def async_update_tracked_entity_ids(
+        self, entity_ids: Iterable[str] | None
```

Comment:
```suggestion
        self, entity_ids: Collection[str] | None
```
You could also consider using `Collection` instead. Something I learned recently, since `Iterable` doesn't include a `__len__` method, `bool(some_iterable)` can always be `True` (in particular for generators).

`Collection` however will require `__len__` to be implemented and thus enable a small shortcut later.

The difference in this case is probably negligible but since the method is called with `sets` anyway, you could also just use it.
https://github.com/home-assistant/core/blob/6c0ad54a84393181e9a4e2af7f984306f2c38d08/homeassistant/components/group/__init__.py#L659-L681